test/integration/apiserver/field_validation_test.go

```diff
@@ -344,14 +345,54 @@ spec:
 	`
 )
 
-// TestFieldValidationPost tests POST requests containing unknown fields with
-// strict and non-strict field validation.
-func TestFieldValidationPost(t *testing.T) {
+func TestFieldValidation(t *testing.T) {
 	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ServerSideFieldValidation, true)()
 
-	_, client, closeFn := setup(t)
-	defer closeFn()
+	server, err := kubeapiservertesting.StartTestServer(t, kubeapiservertesting.NewDefaultTestServerOptions(), nil, framework.SharedEtcd())
+	if err != nil {
+		t.Fatal(err)
+	}
+	config := server.ClientConfig
+	defer server.TearDownFn()
+
+	schemaCRD := setupCRD(t, config, "schema.example.com", false)
+	schemalessCRD := setupCRD(t, config, "schemaless.example.com", true)
```

Comment:
should we pass a DeepCopy() to each test of each crd?
it may be right now but is easy in the future to make a mistake here?

Reply:
that's a good point... the tests actually only need the group/version/kind and group/version/resource info from the CRD... narrowed the inputs to pass that in in a way that can't be mutated by the test